// plugin-welcome/src/api/models/ControllersRoom.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * 
 * @export
 * @interface ControllersRoom
 */
export interface ControllersRoom {
    /**
     * 
     * @type {number}
     * @memberof ControllersRoom
     */
    equipment?: number;
    /**
     * 
     * @type {number}
     * @memberof ControllersRoom
     */
    facility?: number;
    /**
     * 
     * @type {number}
     * @memberof ControllersRoom
     */
    nearbyPlace?: number;
    /**
     * 
     * @type {number}
     * @memberof ControllersRoom
     */
    quantity?: number;
    /**
     * 
     * @type {number}
     * @memberof ControllersRoom
     */
    roomprice?: number;
    /**
     * 
     * @type {string}
     * @memberof ControllersRoom
     */
    roomtypename?: string;
    /**
     * 
     * @type {number}
     * @memberof ControllersRoom
     */
    stayType?: number;
}

export function ControllersRoomFromJSON(json: any): ControllersRoom {
    return ControllersRoomFromJSONTyped(json, false);
}

export function ControllersRoomFromJSONTyped(json: any, ignoreDiscriminator: boolean): ControllersRoom {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'equipment': !exists(json, 'equipment') ? undefined : json['equipment'],
        'facility': !exists(json, 'facility') ? undefined : json['facility'],
        'nearbyPlace': !exists(json, 'nearbyPlace') ? undefined : json['nearbyPlace'],
        'quantity': !exists(json, 'quantity') ? undefined : json['quantity'],
        'roomprice': !exists(json, 'roomprice') ? undefined : json['roomprice'],
        'roomtypename': !exists(json, 'roomtypename') ? undefined : json['roomtypename'],
        'stayType': !exists(json, 'stayType') ? undefined : json['stayType'],
    };
}

export function ControllersRoomToJSON(value?: ControllersRoom | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'equipment': value.equipment,
        'facility': value.facility,
        'nearbyPlace': value.nearbyPlace,
        'quantity': value.quantity,
        'roomprice': value.roomprice,
        'roomtypename': value.roomtypename,
        'stayType': value.stayType,
    };
}
