// plugin-welcome/src/api/models/ControllersCleaningroom.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * 
 * @export
 * @interface ControllersCleaningroom
 */
export interface ControllersCleaningroom {
    /**
     * 
     * @type {number}
     * @memberof ControllersCleaningroom
     */
    cleanername?: number;
    /**
     * 
     * @type {string}
     * @memberof ControllersCleaningroom
     */
    dateandstarttime?: string;
    /**
     * 
     * @type {number}
     * @memberof ControllersCleaningroom
     */
    employee?: number;
    /**
     * 
     * @type {number}
     * @memberof ControllersCleaningroom
     */
    lengthtime?: number;
    /**
     * 
     * @type {string}
     * @memberof ControllersCleaningroom
     */
    note?: string;
    /**
     * 
     * @type {number}
     * @memberof ControllersCleaningroom
     */
    numofem?: number;
    /**
     * 
     * @type {string}
     * @memberof ControllersCleaningroom
     */
    phonenumber?: string;
    /**
     * 
     * @type {number}
     * @memberof ControllersCleaningroom
     */
    roomdetail?: number;
}

export function ControllersCleaningroomFromJSON(json: any): ControllersCleaningroom {
    return ControllersCleaningroomFromJSONTyped(json, false);
}

export function ControllersCleaningroomFromJSONTyped(json: any, ignoreDiscriminator: boolean): ControllersCleaningroom {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'cleanername': !exists(json, 'cleanername') ? undefined : json['cleanername'],
        'dateandstarttime': !exists(json, 'dateandstarttime') ? undefined : json['dateandstarttime'],
        'employee': !exists(json, 'employee') ? undefined : json['employee'],
        'lengthtime': !exists(json, 'lengthtime') ? undefined : json['lengthtime'],
        'note': !exists(json, 'note') ? undefined : json['note'],
        'numofem': !exists(json, 'numofem') ? undefined : json['numofem'],
        'phonenumber': !exists(json, 'phonenumber') ? undefined : json['phonenumber'],
        'roomdetail': !exists(json, 'roomdetail') ? undefined : json['roomdetail'],
    };
}

export function ControllersCleaningroomToJSON(value?: ControllersCleaningroom | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'cleanername': value.cleanername,
        'dateandstarttime': value.dateandstarttime,
        'employee': value.employee,
        'lengthtime': value.lengthtime,
        'note': value.note,
        'numofem': value.numofem,
        'phonenumber': value.phonenumber,
        'roomdetail': value.roomdetail,
    };
}
