// plugin-welcome/src/api/models/EntBill.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntBillEdges,
    EntBillEdgesFromJSON,
    EntBillEdgesFromJSONTyped,
    EntBillEdgesToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntBill
 */
export interface EntBill {
    /**
     * Addedtime holds the value of the "addedtime" field.
     * @type {string}
     * @memberof EntBill
     */
    addedtime?: string;
    /**
     * 
     * @type {EntBillEdges}
     * @memberof EntBill
     */
    edges?: EntBillEdges;
    /**
     * ID of the ent.
     * @type {number}
     * @memberof EntBill
     */
    id?: number;
    /**
     * Taxpayer holds the value of the "taxpayer" field.
     * @type {string}
     * @memberof EntBill
     */
    taxpayer?: string;
    /**
     * Tell holds the value of the "tell" field.
     * @type {string}
     * @memberof EntBill
     */
    tell?: string;
    /**
     * Total holds the value of the "total" field.
     * @type {string}
     * @memberof EntBill
     */
    total?: string;
}

export function EntBillFromJSON(json: any): EntBill {
    return EntBillFromJSONTyped(json, false);
}

export function EntBillFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntBill {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'addedtime': !exists(json, 'addedtime') ? undefined : json['addedtime'],
        'edges': !exists(json, 'edges') ? undefined : EntBillEdgesFromJSON(json['edges']),
        'id': !exists(json, 'id') ? undefined : json['id'],
        'taxpayer': !exists(json, 'taxpayer') ? undefined : json['taxpayer'],
        'tell': !exists(json, 'tell') ? undefined : json['tell'],
        'total': !exists(json, 'total') ? undefined : json['total'],
    };
}

export function EntBillToJSON(value?: EntBill | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'addedtime': value.addedtime,
        'edges': EntBillEdgesToJSON(value.edges),
        'id': value.id,
        'taxpayer': value.taxpayer,
        'tell': value.tell,
        'total': value.total,
    };
}
