// plugin-welcome/src/api/models/EntBillEdges.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntLease,
    EntLeaseFromJSON,
    EntLeaseFromJSONTyped,
    EntLeaseToJSON,
    EntPayment,
    EntPaymentFromJSON,
    EntPaymentFromJSONTyped,
    EntPaymentToJSON,
    EntSituation,
    EntSituationFromJSON,
    EntSituationFromJSONTyped,
    EntSituationToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntBillEdges
 */
export interface EntBillEdges {
    /**
     * 
     * @type {EntLease}
     * @memberof EntBillEdges
     */
    lease?: EntLease;
    /**
     * 
     * @type {EntPayment}
     * @memberof EntBillEdges
     */
    payment?: EntPayment;
    /**
     * 
     * @type {EntSituation}
     * @memberof EntBillEdges
     */
    situation?: EntSituation;
}

export function EntBillEdgesFromJSON(json: any): EntBillEdges {
    return EntBillEdgesFromJSONTyped(json, false);
}

export function EntBillEdgesFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntBillEdges {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'lease': !exists(json, 'lease') ? undefined : EntLeaseFromJSON(json['lease']),
        'payment': !exists(json, 'payment') ? undefined : EntPaymentFromJSON(json['payment']),
        'situation': !exists(json, 'situation') ? undefined : EntSituationFromJSON(json['situation']),
    };
}

export function EntBillEdgesToJSON(value?: EntBillEdges | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'lease': EntLeaseToJSON(value.lease),
        'payment': EntPaymentToJSON(value.payment),
        'situation': EntSituationToJSON(value.situation),
    };
}
