// plugin-welcome/src/api/models/EntCleanerNameEdges.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntCleaningRoom,
    EntCleaningRoomFromJSON,
    EntCleaningRoomFromJSONTyped,
    EntCleaningRoomToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntCleanerNameEdges
 */
export interface EntCleanerNameEdges {
    /**
     * Cleaningrooms holds the value of the cleaningrooms edge.
     * @type {Array<EntCleaningRoom>}
     * @memberof EntCleanerNameEdges
     */
    cleaningrooms?: Array<EntCleaningRoom>;
}

export function EntCleanerNameEdgesFromJSON(json: any): EntCleanerNameEdges {
    return EntCleanerNameEdgesFromJSONTyped(json, false);
}

export function EntCleanerNameEdgesFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntCleanerNameEdges {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'cleaningrooms': !exists(json, 'cleaningrooms') ? undefined : ((json['cleaningrooms'] as Array<any>).map(EntCleaningRoomFromJSON)),
    };
}

export function EntCleanerNameEdgesToJSON(value?: EntCleanerNameEdges | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'cleaningrooms': value.cleaningrooms === undefined ? undefined : ((value.cleaningrooms as Array<any>).map(EntCleaningRoomToJSON)),
    };
}
