// plugin-welcome/src/api/models/EntCleaningRoomEdges.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntCleanerName,
    EntCleanerNameFromJSON,
    EntCleanerNameFromJSONTyped,
    EntCleanerNameToJSON,
    EntLengthTime,
    EntLengthTimeFromJSON,
    EntLengthTimeFromJSONTyped,
    EntLengthTimeToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntCleaningRoomEdges
 */
export interface EntCleaningRoomEdges {
    /**
     * 
     * @type {EntCleanerName}
     * @memberof EntCleaningRoomEdges
     */
    cleanerName?: EntCleanerName;
    /**
     * 
     * @type {EntLengthTime}
     * @memberof EntCleaningRoomEdges
     */
    lengthTime?: EntLengthTime;
}

export function EntCleaningRoomEdgesFromJSON(json: any): EntCleaningRoomEdges {
    return EntCleaningRoomEdgesFromJSONTyped(json, false);
}

export function EntCleaningRoomEdgesFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntCleaningRoomEdges {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'cleanerName': !exists(json, 'cleanerName') ? undefined : EntCleanerNameFromJSON(json['cleanerName']),
        'lengthTime': !exists(json, 'lengthTime') ? undefined : EntLengthTimeFromJSON(json['lengthTime']),
    };
}

export function EntCleaningRoomEdgesToJSON(value?: EntCleaningRoomEdges | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'cleanerName': EntCleanerNameToJSON(value.cleanerName),
        'lengthTime': EntLengthTimeToJSON(value.lengthTime),
    };
}
