// plugin-welcome/src/api/models/EntCleanerName.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntCleanernameEdges,
    EntCleanernameEdgesFromJSON,
    EntCleanernameEdgesFromJSONTyped,
    EntCleanernameEdgesToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntCleanername
 */
export interface EntCleanername {
    /**
     * Cleanername holds the value of the "cleanername" field.
     * @type {string}
     * @memberof EntCleanername
     */
    cleanername?: string;
    /**
     * 
     * @type {EntCleanernameEdges}
     * @memberof EntCleanername
     */
    edges?: EntCleanernameEdges;
    /**
     * ID of the ent.
     * @type {number}
     * @memberof EntCleanername
     */
    id?: number;
}

export function EntCleanernameFromJSON(json: any): EntCleanername {
    return EntCleanernameFromJSONTyped(json, false);
}

export function EntCleanernameFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntCleanername {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'cleanername': !exists(json, 'cleanername') ? undefined : json['cleanername'],
        'edges': !exists(json, 'edges') ? undefined : EntCleanernameEdgesFromJSON(json['edges']),
        'id': !exists(json, 'id') ? undefined : json['id'],
    };
}

export function EntCleanernameToJSON(value?: EntCleanername | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'cleanername': value.cleanername,
        'edges': EntCleanernameEdgesToJSON(value.edges),
        'id': value.id,
    };
}
