// plugin-welcome/src/api/models/EntCleaningRoomEdges.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntCleanername,
    EntCleanernameFromJSON,
    EntCleanernameFromJSONTyped,
    EntCleanernameToJSON,
    EntEmployee,
    EntEmployeeFromJSON,
    EntEmployeeFromJSONTyped,
    EntEmployeeToJSON,
    EntLengthtime,
    EntLengthtimeFromJSON,
    EntLengthtimeFromJSONTyped,
    EntLengthtimeToJSON,
    EntRoomdetail,
    EntRoomdetailFromJSON,
    EntRoomdetailFromJSONTyped,
    EntRoomdetailToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntCleaningroomEdges
 */
export interface EntCleaningroomEdges {
    /**
     * 
     * @type {EntCleanername}
     * @memberof EntCleaningroomEdges
     */
    cleanername?: EntCleanername;
    /**
     * 
     * @type {EntEmployee}
     * @memberof EntCleaningroomEdges
     */
    employee?: EntEmployee;
    /**
     * 
     * @type {EntLengthtime}
     * @memberof EntCleaningroomEdges
     */
    lengthtime?: EntLengthtime;
    /**
     * 
     * @type {EntRoomdetail}
     * @memberof EntCleaningroomEdges
     */
    roomdetail?: EntRoomdetail;
}

export function EntCleaningroomEdgesFromJSON(json: any): EntCleaningroomEdges {
    return EntCleaningroomEdgesFromJSONTyped(json, false);
}

export function EntCleaningroomEdgesFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntCleaningroomEdges {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'cleanername': !exists(json, 'cleanername') ? undefined : EntCleanernameFromJSON(json['cleanername']),
        'employee': !exists(json, 'employee') ? undefined : EntEmployeeFromJSON(json['employee']),
        'lengthtime': !exists(json, 'lengthtime') ? undefined : EntLengthtimeFromJSON(json['lengthtime']),
        'roomdetail': !exists(json, 'roomdetail') ? undefined : EntRoomdetailFromJSON(json['roomdetail']),
    };
}

export function EntCleaningroomEdgesToJSON(value?: EntCleaningroomEdges | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'cleanername': EntCleanernameToJSON(value.cleanername),
        'employee': EntEmployeeToJSON(value.employee),
        'lengthtime': EntLengthtimeToJSON(value.lengthtime),
        'roomdetail': EntRoomdetailToJSON(value.roomdetail),
    };
}
