// plugin-welcome/src/api/models/EntCleaningRoom.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntCleaningroomEdges,
    EntCleaningroomEdgesFromJSON,
    EntCleaningroomEdgesFromJSONTyped,
    EntCleaningroomEdgesToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntCleaningroom
 */
export interface EntCleaningroom {
    /**
     * Dateandstarttime holds the value of the "dateandstarttime" field.
     * @type {string}
     * @memberof EntCleaningroom
     */
    dateandstarttime?: string;
    /**
     * 
     * @type {EntCleaningroomEdges}
     * @memberof EntCleaningroom
     */
    edges?: EntCleaningroomEdges;
    /**
     * ID of the ent.
     * @type {number}
     * @memberof EntCleaningroom
     */
    id?: number;
    /**
     * Note holds the value of the "note" field.
     * @type {string}
     * @memberof EntCleaningroom
     */
    note?: string;
    /**
     * Numofem holds the value of the "numofem" field.
     * @type {number}
     * @memberof EntCleaningroom
     */
    numofem?: number;
    /**
     * Phonenumber holds the value of the "phonenumber" field.
     * @type {string}
     * @memberof EntCleaningroom
     */
    phonenumber?: string;
}

export function EntCleaningroomFromJSON(json: any): EntCleaningroom {
    return EntCleaningroomFromJSONTyped(json, false);
}

export function EntCleaningroomFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntCleaningroom {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'dateandstarttime': !exists(json, 'dateandstarttime') ? undefined : json['dateandstarttime'],
        'edges': !exists(json, 'edges') ? undefined : EntCleaningroomEdgesFromJSON(json['edges']),
        'id': !exists(json, 'id') ? undefined : json['id'],
        'note': !exists(json, 'note') ? undefined : json['note'],
        'numofem': !exists(json, 'numofem') ? undefined : json['numofem'],
        'phonenumber': !exists(json, 'phonenumber') ? undefined : json['phonenumber'],
    };
}

export function EntCleaningroomToJSON(value?: EntCleaningroom | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'dateandstarttime': value.dateandstarttime,
        'edges': EntCleaningroomEdgesToJSON(value.edges),
        'id': value.id,
        'note': value.note,
        'numofem': value.numofem,
        'phonenumber': value.phonenumber,
    };
}
