// plugin-welcome/src/api/models/EntEquipmentEdges.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntRoomdetail,
    EntRoomdetailFromJSON,
    EntRoomdetailFromJSONTyped,
    EntRoomdetailToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntEquipmentEdges
 */
export interface EntEquipmentEdges {
    /**
     * 
     * @type {EntRoomdetail}
     * @memberof EntEquipmentEdges
     */
    roomdetail?: EntRoomdetail;
}

export function EntEquipmentEdgesFromJSON(json: any): EntEquipmentEdges {
    return EntEquipmentEdgesFromJSONTyped(json, false);
}

export function EntEquipmentEdgesFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntEquipmentEdges {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'roomdetail': !exists(json, 'roomdetail') ? undefined : EntRoomdetailFromJSON(json['roomdetail']),
    };
}

export function EntEquipmentEdgesToJSON(value?: EntEquipmentEdges | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'roomdetail': EntRoomdetailToJSON(value.roomdetail),
    };
}
