// plugin-welcome/src/api/models/EntEquipment.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntEquipmentEdges,
    EntEquipmentEdgesFromJSON,
    EntEquipmentEdgesFromJSONTyped,
    EntEquipmentEdgesToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntEquipment
 */
export interface EntEquipment {
    /**
     * 
     * @type {EntEquipmentEdges}
     * @memberof EntEquipment
     */
    edges?: EntEquipmentEdges;
    /**
     * Equipment holds the value of the "equipment" field.
     * @type {string}
     * @memberof EntEquipment
     */
    equipment?: string;
    /**
     * ID of the ent.
     * @type {number}
     * @memberof EntEquipment
     */
    id?: number;
}

export function EntEquipmentFromJSON(json: any): EntEquipment {
    return EntEquipmentFromJSONTyped(json, false);
}

export function EntEquipmentFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntEquipment {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'edges': !exists(json, 'edges') ? undefined : EntEquipmentEdgesFromJSON(json['edges']),
        'equipment': !exists(json, 'equipment') ? undefined : json['equipment'],
        'id': !exists(json, 'id') ? undefined : json['id'],
    };
}

export function EntEquipmentToJSON(value?: EntEquipment | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'edges': EntEquipmentEdgesToJSON(value.edges),
        'equipment': value.equipment,
        'id': value.id,
    };
}
