// plugin-welcome/src/api/models/EntEmployee.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntEmployeeEdges,
    EntEmployeeEdgesFromJSON,
    EntEmployeeEdgesFromJSONTyped,
    EntEmployeeEdgesToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntEmployee
 */
export interface EntEmployee {
    /**
     * 
     * @type {EntEmployeeEdges}
     * @memberof EntEmployee
     */
    edges?: EntEmployeeEdges;
    /**
     * Email holds the value of the "email" field.
     * @type {string}
     * @memberof EntEmployee
     */
    email?: string;
    /**
     * ID of the ent.
     * @type {number}
     * @memberof EntEmployee
     */
    id?: number;
    /**
     * Name holds the value of the "name" field.
     * @type {string}
     * @memberof EntEmployee
     */
    name?: string;
    /**
     * Password holds the value of the "password" field.
     * @type {string}
     * @memberof EntEmployee
     */
    password?: string;
}

export function EntEmployeeFromJSON(json: any): EntEmployee {
    return EntEmployeeFromJSONTyped(json, false);
}

export function EntEmployeeFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntEmployee {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'edges': !exists(json, 'edges') ? undefined : EntEmployeeEdgesFromJSON(json['edges']),
        'email': !exists(json, 'email') ? undefined : json['email'],
        'id': !exists(json, 'id') ? undefined : json['id'],
        'name': !exists(json, 'name') ? undefined : json['name'],
        'password': !exists(json, 'password') ? undefined : json['password'],
    };
}

export function EntEmployeeToJSON(value?: EntEmployee | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'edges': EntEmployeeEdgesToJSON(value.edges),
        'email': value.email,
        'id': value.id,
        'name': value.name,
        'password': value.password,
    };
}
