// plugin-welcome/src/api/models/EntJobposition.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntJobpositionEdges,
    EntJobpositionEdgesFromJSON,
    EntJobpositionEdgesFromJSONTyped,
    EntJobpositionEdgesToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntJobposition
 */
export interface EntJobposition {
    /**
     * 
     * @type {EntJobpositionEdges}
     * @memberof EntJobposition
     */
    edges?: EntJobpositionEdges;
    /**
     * ID of the ent.
     * @type {number}
     * @memberof EntJobposition
     */
    id?: number;
    /**
     * Positionname holds the value of the "positionname" field.
     * @type {string}
     * @memberof EntJobposition
     */
    positionname?: string;
}

export function EntJobpositionFromJSON(json: any): EntJobposition {
    return EntJobpositionFromJSONTyped(json, false);
}

export function EntJobpositionFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntJobposition {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'edges': !exists(json, 'edges') ? undefined : EntJobpositionEdgesFromJSON(json['edges']),
        'id': !exists(json, 'id') ? undefined : json['id'],
        'positionname': !exists(json, 'positionname') ? undefined : json['positionname'],
    };
}

export function EntJobpositionToJSON(value?: EntJobposition | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'edges': EntJobpositionEdgesToJSON(value.edges),
        'id': value.id,
        'positionname': value.positionname,
    };
}
