// plugin-welcome/src/api/models/EntLengthTime.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntLengthTimeEdges,
    EntLengthTimeEdgesFromJSON,
    EntLengthTimeEdgesFromJSONTyped,
    EntLengthTimeEdgesToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntLengthTime
 */
export interface EntLengthTime {
    /**
     * 
     * @type {EntLengthTimeEdges}
     * @memberof EntLengthTime
     */
    edges?: EntLengthTimeEdges;
    /**
     * ID of the ent.
     * @type {number}
     * @memberof EntLengthTime
     */
    id?: number;
    /**
     * Lengthtime holds the value of the "lengthtime" field.
     * @type {string}
     * @memberof EntLengthTime
     */
    lengthtime?: string;
}

export function EntLengthTimeFromJSON(json: any): EntLengthTime {
    return EntLengthTimeFromJSONTyped(json, false);
}

export function EntLengthTimeFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntLengthTime {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'edges': !exists(json, 'edges') ? undefined : EntLengthTimeEdgesFromJSON(json['edges']),
        'id': !exists(json, 'id') ? undefined : json['id'],
        'lengthtime': !exists(json, 'lengthtime') ? undefined : json['lengthtime'],
    };
}

export function EntLengthTimeToJSON(value?: EntLengthTime | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'edges': EntLengthTimeEdgesToJSON(value.edges),
        'id': value.id,
        'lengthtime': value.lengthtime,
    };
}
