// plugin-welcome/src/api/models/EntLease.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntLeaseEdges,
    EntLeaseEdgesFromJSON,
    EntLeaseEdgesFromJSONTyped,
    EntLeaseEdgesToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntLease
 */
export interface EntLease {
    /**
     * Addedtime holds the value of the "addedtime" field.
     * @type {string}
     * @memberof EntLease
     */
    addedtime?: string;
    /**
     * 
     * @type {EntLeaseEdges}
     * @memberof EntLease
     */
    edges?: EntLeaseEdges;
    /**
     * ID of the ent.
     * @type {number}
     * @memberof EntLease
     */
    id?: number;
    /**
     * Numbtenant holds the value of the "numbtenant" field.
     * @type {string}
     * @memberof EntLease
     */
    numbtenant?: string;
    /**
     * Pettenant holds the value of the "pettenant" field.
     * @type {string}
     * @memberof EntLease
     */
    pettenant?: string;
    /**
     * Tenant holds the value of the "tenant" field.
     * @type {string}
     * @memberof EntLease
     */
    tenant?: string;
}

export function EntLeaseFromJSON(json: any): EntLease {
    return EntLeaseFromJSONTyped(json, false);
}

export function EntLeaseFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntLease {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'addedtime': !exists(json, 'addedtime') ? undefined : json['addedtime'],
        'edges': !exists(json, 'edges') ? undefined : EntLeaseEdgesFromJSON(json['edges']),
        'id': !exists(json, 'id') ? undefined : json['id'],
        'numbtenant': !exists(json, 'numbtenant') ? undefined : json['numbtenant'],
        'pettenant': !exists(json, 'pettenant') ? undefined : json['pettenant'],
        'tenant': !exists(json, 'tenant') ? undefined : json['tenant'],
    };
}

export function EntLeaseToJSON(value?: EntLease | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'addedtime': value.addedtime,
        'edges': EntLeaseEdgesToJSON(value.edges),
        'id': value.id,
        'numbtenant': value.numbtenant,
        'pettenant': value.pettenant,
        'tenant': value.tenant,
    };
}
