// plugin-welcome/src/api/models/EntLengthTime.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntLengthtimeEdges,
    EntLengthtimeEdgesFromJSON,
    EntLengthtimeEdgesFromJSONTyped,
    EntLengthtimeEdgesToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntLengthtime
 */
export interface EntLengthtime {
    /**
     * 
     * @type {EntLengthtimeEdges}
     * @memberof EntLengthtime
     */
    edges?: EntLengthtimeEdges;
    /**
     * ID of the ent.
     * @type {number}
     * @memberof EntLengthtime
     */
    id?: number;
    /**
     * Lengthtime holds the value of the "lengthtime" field.
     * @type {string}
     * @memberof EntLengthtime
     */
    lengthtime?: string;
}

export function EntLengthtimeFromJSON(json: any): EntLengthtime {
    return EntLengthtimeFromJSONTyped(json, false);
}

export function EntLengthtimeFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntLengthtime {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'edges': !exists(json, 'edges') ? undefined : EntLengthtimeEdgesFromJSON(json['edges']),
        'id': !exists(json, 'id') ? undefined : json['id'],
        'lengthtime': !exists(json, 'lengthtime') ? undefined : json['lengthtime'],
    };
}

export function EntLengthtimeToJSON(value?: EntLengthtime | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'edges': EntLengthtimeEdgesToJSON(value.edges),
        'id': value.id,
        'lengthtime': value.lengthtime,
    };
}
