// plugin-welcome/src/api/models/EntLeaseEdges.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntBill,
    EntBillFromJSON,
    EntBillFromJSONTyped,
    EntBillToJSON,
    EntDeposit,
    EntDepositFromJSON,
    EntDepositFromJSONTyped,
    EntDepositToJSON,
    EntEmployee,
    EntEmployeeFromJSON,
    EntEmployeeFromJSONTyped,
    EntEmployeeToJSON,
    EntRepairinvoice,
    EntRepairinvoiceFromJSON,
    EntRepairinvoiceFromJSONTyped,
    EntRepairinvoiceToJSON,
    EntRoomdetail,
    EntRoomdetailFromJSON,
    EntRoomdetailFromJSONTyped,
    EntRoomdetailToJSON,
    EntWifi,
    EntWifiFromJSON,
    EntWifiFromJSONTyped,
    EntWifiToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntLeaseEdges
 */
export interface EntLeaseEdges {
    /**
     * Bill holds the value of the bill edge.
     * @type {Array<EntBill>}
     * @memberof EntLeaseEdges
     */
    bill?: Array<EntBill>;
    /**
     * 
     * @type {EntEmployee}
     * @memberof EntLeaseEdges
     */
    employee?: EntEmployee;
    /**
     * Leases holds the value of the leases edge.
     * @type {Array<EntDeposit>}
     * @memberof EntLeaseEdges
     */
    leases?: Array<EntDeposit>;
    /**
     * Repairinvoices holds the value of the repairinvoices edge.
     * @type {Array<EntRepairinvoice>}
     * @memberof EntLeaseEdges
     */
    repairinvoices?: Array<EntRepairinvoice>;
    /**
     * 
     * @type {EntRoomdetail}
     * @memberof EntLeaseEdges
     */
    roomdetail?: EntRoomdetail;
    /**
     * 
     * @type {EntWifi}
     * @memberof EntLeaseEdges
     */
    wifi?: EntWifi;
}

export function EntLeaseEdgesFromJSON(json: any): EntLeaseEdges {
    return EntLeaseEdgesFromJSONTyped(json, false);
}

export function EntLeaseEdgesFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntLeaseEdges {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'bill': !exists(json, 'Bill') ? undefined : ((json['Bill'] as Array<any>).map(EntBillFromJSON)),
        'employee': !exists(json, 'Employee') ? undefined : EntEmployeeFromJSON(json['Employee']),
        'leases': !exists(json, 'Leases') ? undefined : ((json['Leases'] as Array<any>).map(EntDepositFromJSON)),
        'repairinvoices': !exists(json, 'Repairinvoices') ? undefined : ((json['Repairinvoices'] as Array<any>).map(EntRepairinvoiceFromJSON)),
        'roomdetail': !exists(json, 'Roomdetail') ? undefined : EntRoomdetailFromJSON(json['Roomdetail']),
        'wifi': !exists(json, 'Wifi') ? undefined : EntWifiFromJSON(json['Wifi']),
    };
}

export function EntLeaseEdgesToJSON(value?: EntLeaseEdges | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'bill': value.bill === undefined ? undefined : ((value.bill as Array<any>).map(EntBillToJSON)),
        'employee': EntEmployeeToJSON(value.employee),
        'leases': value.leases === undefined ? undefined : ((value.leases as Array<any>).map(EntDepositToJSON)),
        'repairinvoices': value.repairinvoices === undefined ? undefined : ((value.repairinvoices as Array<any>).map(EntRepairinvoiceToJSON)),
        'roomdetail': EntRoomdetailToJSON(value.roomdetail),
        'wifi': EntWifiToJSON(value.wifi),
    };
}
