// plugin-welcome/src/api/models/EntPetrule.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntPetruleEdges,
    EntPetruleEdgesFromJSON,
    EntPetruleEdgesFromJSONTyped,
    EntPetruleEdgesToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntPetrule
 */
export interface EntPetrule {
    /**
     * 
     * @type {EntPetruleEdges}
     * @memberof EntPetrule
     */
    edges?: EntPetruleEdges;
    /**
     * ID of the ent.
     * @type {number}
     * @memberof EntPetrule
     */
    id?: number;
    /**
     * Petrule holds the value of the "petrule" field.
     * @type {string}
     * @memberof EntPetrule
     */
    petrule?: string;
}

export function EntPetruleFromJSON(json: any): EntPetrule {
    return EntPetruleFromJSONTyped(json, false);
}

export function EntPetruleFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntPetrule {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'edges': !exists(json, 'edges') ? undefined : EntPetruleEdgesFromJSON(json['edges']),
        'id': !exists(json, 'id') ? undefined : json['id'],
        'petrule': !exists(json, 'petrule') ? undefined : json['petrule'],
    };
}

export function EntPetruleToJSON(value?: EntPetrule | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'edges': EntPetruleEdgesToJSON(value.edges),
        'id': value.id,
        'petrule': value.petrule,
    };
}
