// plugin-welcome/src/api/models/EntPledge.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntPledgeEdges,
    EntPledgeEdgesFromJSON,
    EntPledgeEdgesFromJSONTyped,
    EntPledgeEdgesToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntPledge
 */
export interface EntPledge {
    /**
     * 
     * @type {EntPledgeEdges}
     * @memberof EntPledge
     */
    edges?: EntPledgeEdges;
    /**
     * ID of the ent.
     * @type {number}
     * @memberof EntPledge
     */
    id?: number;
    /**
     * Provision holds the value of the "provision" field.
     * @type {string}
     * @memberof EntPledge
     */
    provision?: string;
}

export function EntPledgeFromJSON(json: any): EntPledge {
    return EntPledgeFromJSONTyped(json, false);
}

export function EntPledgeFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntPledge {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'edges': !exists(json, 'edges') ? undefined : EntPledgeEdgesFromJSON(json['edges']),
        'id': !exists(json, 'id') ? undefined : json['id'],
        'provision': !exists(json, 'provision') ? undefined : json['provision'],
    };
}

export function EntPledgeToJSON(value?: EntPledge | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'edges': EntPledgeEdgesToJSON(value.edges),
        'id': value.id,
        'provision': value.provision,
    };
}
