// plugin-welcome/src/api/models/EntPaymentEdges.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntBill,
    EntBillFromJSON,
    EntBillFromJSONTyped,
    EntBillToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntPaymentEdges
 */
export interface EntPaymentEdges {
    /**
     * Payments holds the value of the payments edge.
     * @type {Array<EntBill>}
     * @memberof EntPaymentEdges
     */
    payments?: Array<EntBill>;
}

export function EntPaymentEdgesFromJSON(json: any): EntPaymentEdges {
    return EntPaymentEdgesFromJSONTyped(json, false);
}

export function EntPaymentEdgesFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntPaymentEdges {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'payments': !exists(json, 'payments') ? undefined : ((json['payments'] as Array<any>).map(EntBillFromJSON)),
    };
}

export function EntPaymentEdgesToJSON(value?: EntPaymentEdges | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'payments': value.payments === undefined ? undefined : ((value.payments as Array<any>).map(EntBillToJSON)),
    };
}
