// plugin-welcome/src/api/models/EntRoomdetailEdges.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntEquipment,
    EntEquipmentFromJSON,
    EntEquipmentFromJSONTyped,
    EntEquipmentToJSON,
    EntFacilitie,
    EntFacilitieFromJSON,
    EntFacilitieFromJSONTyped,
    EntFacilitieToJSON,
    EntNearbyplace,
    EntNearbyplaceFromJSON,
    EntNearbyplaceFromJSONTyped,
    EntNearbyplaceToJSON,
    EntQuantity,
    EntQuantityFromJSON,
    EntQuantityFromJSONTyped,
    EntQuantityToJSON,
    EntStaytype,
    EntStaytypeFromJSON,
    EntStaytypeFromJSONTyped,
    EntStaytypeToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntRoomdetailEdges
 */
export interface EntRoomdetailEdges {
    /**
     * Equipments holds the value of the equipments edge.
     * @type {Array<EntEquipment>}
     * @memberof EntRoomdetailEdges
     */
    equipments?: Array<EntEquipment>;
    /**
     * Facilities holds the value of the facilities edge.
     * @type {Array<EntFacilitie>}
     * @memberof EntRoomdetailEdges
     */
    facilities?: Array<EntFacilitie>;
    /**
     * Nearbyplaces holds the value of the nearbyplaces edge.
     * @type {Array<EntNearbyplace>}
     * @memberof EntRoomdetailEdges
     */
    nearbyplaces?: Array<EntNearbyplace>;
    /**
     * 
     * @type {EntQuantity}
     * @memberof EntRoomdetailEdges
     */
    quantity?: EntQuantity;
    /**
     * 
     * @type {EntStaytype}
     * @memberof EntRoomdetailEdges
     */
    staytype?: EntStaytype;
}

export function EntRoomdetailEdgesFromJSON(json: any): EntRoomdetailEdges {
    return EntRoomdetailEdgesFromJSONTyped(json, false);
}

export function EntRoomdetailEdgesFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntRoomdetailEdges {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'equipments': !exists(json, 'equipments') ? undefined : ((json['equipments'] as Array<any>).map(EntEquipmentFromJSON)),
        'facilities': !exists(json, 'facilities') ? undefined : ((json['facilities'] as Array<any>).map(EntFacilitieFromJSON)),
        'nearbyplaces': !exists(json, 'nearbyplaces') ? undefined : ((json['nearbyplaces'] as Array<any>).map(EntNearbyplaceFromJSON)),
        'quantity': !exists(json, 'quantity') ? undefined : EntQuantityFromJSON(json['quantity']),
        'staytype': !exists(json, 'staytype') ? undefined : EntStaytypeFromJSON(json['staytype']),
    };
}

export function EntRoomdetailEdgesToJSON(value?: EntRoomdetailEdges | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'equipments': value.equipments === undefined ? undefined : ((value.equipments as Array<any>).map(EntEquipmentToJSON)),
        'facilities': value.facilities === undefined ? undefined : ((value.facilities as Array<any>).map(EntFacilitieToJSON)),
        'nearbyplaces': value.nearbyplaces === undefined ? undefined : ((value.nearbyplaces as Array<any>).map(EntNearbyplaceToJSON)),
        'quantity': EntQuantityToJSON(value.quantity),
        'staytype': EntStaytypeToJSON(value.staytype),
    };
}
