// plugin-welcome/src/api/models/EntRoomdetail.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntRoomdetailEdges,
    EntRoomdetailEdgesFromJSON,
    EntRoomdetailEdgesFromJSONTyped,
    EntRoomdetailEdgesToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntRoomdetail
 */
export interface EntRoomdetail {
    /**
     * Bed holds the value of the "bed" field.
     * @type {number}
     * @memberof EntRoomdetail
     */
    bed?: number;
    /**
     * 
     * @type {EntRoomdetailEdges}
     * @memberof EntRoomdetail
     */
    edges?: EntRoomdetailEdges;
    /**
     * ID of the ent.
     * @type {number}
     * @memberof EntRoomdetail
     */
    id?: number;
    /**
     * Phone holds the value of the "phone" field.
     * @type {string}
     * @memberof EntRoomdetail
     */
    phone?: string;
    /**
     * Roomnumber holds the value of the "roomnumber" field.
     * @type {string}
     * @memberof EntRoomdetail
     */
    roomnumber?: string;
    /**
     * Roomprice holds the value of the "roomprice" field.
     * @type {string}
     * @memberof EntRoomdetail
     */
    roomprice?: string;
    /**
     * Roomtypename holds the value of the "roomtypename" field.
     * @type {string}
     * @memberof EntRoomdetail
     */
    roomtypename?: string;
    /**
     * Sleep holds the value of the "sleep" field.
     * @type {number}
     * @memberof EntRoomdetail
     */
    sleep?: number;
}

export function EntRoomdetailFromJSON(json: any): EntRoomdetail {
    return EntRoomdetailFromJSONTyped(json, false);
}

export function EntRoomdetailFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntRoomdetail {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'bed': !exists(json, 'bed') ? undefined : json['bed'],
        'edges': !exists(json, 'edges') ? undefined : EntRoomdetailEdgesFromJSON(json['edges']),
        'id': !exists(json, 'id') ? undefined : json['id'],
        'phone': !exists(json, 'phone') ? undefined : json['phone'],
        'roomnumber': !exists(json, 'roomnumber') ? undefined : json['roomnumber'],
        'roomprice': !exists(json, 'roomprice') ? undefined : json['roomprice'],
        'roomtypename': !exists(json, 'roomtypename') ? undefined : json['roomtypename'],
        'sleep': !exists(json, 'sleep') ? undefined : json['sleep'],
    };
}

export function EntRoomdetailToJSON(value?: EntRoomdetail | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'bed': value.bed,
        'edges': EntRoomdetailEdgesToJSON(value.edges),
        'id': value.id,
        'phone': value.phone,
        'roomnumber': value.roomnumber,
        'roomprice': value.roomprice,
        'roomtypename': value.roomtypename,
        'sleep': value.sleep,
    };
}
