// plugin-welcome/src/api/models/EntRentalstatusEdges.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntRepairinvoice,
    EntRepairinvoiceFromJSON,
    EntRepairinvoiceFromJSONTyped,
    EntRepairinvoiceToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntRentalstatusEdges
 */
export interface EntRentalstatusEdges {
    /**
     * Repairinvoices holds the value of the repairinvoices edge.
     * @type {Array<EntRepairinvoice>}
     * @memberof EntRentalstatusEdges
     */
    repairinvoices?: Array<EntRepairinvoice>;
}

export function EntRentalstatusEdgesFromJSON(json: any): EntRentalstatusEdges {
    return EntRentalstatusEdgesFromJSONTyped(json, false);
}

export function EntRentalstatusEdgesFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntRentalstatusEdges {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'repairinvoices': !exists(json, 'Repairinvoices') ? undefined : ((json['Repairinvoices'] as Array<any>).map(EntRepairinvoiceFromJSON)),
    };
}

export function EntRentalstatusEdgesToJSON(value?: EntRentalstatusEdges | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'repairinvoices': value.repairinvoices === undefined ? undefined : ((value.repairinvoices as Array<any>).map(EntRepairinvoiceToJSON)),
    };
}
