// plugin-welcome/src/api/models/EntRoomdetailEdges.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntEmployee,
    EntEmployeeFromJSON,
    EntEmployeeFromJSONTyped,
    EntEmployeeToJSON,
    EntEquipment,
    EntEquipmentFromJSON,
    EntEquipmentFromJSONTyped,
    EntEquipmentToJSON,
    EntFacilitie,
    EntFacilitieFromJSON,
    EntFacilitieFromJSONTyped,
    EntFacilitieToJSON,
    EntLease,
    EntLeaseFromJSON,
    EntLeaseFromJSONTyped,
    EntLeaseToJSON,
    EntNearbyplace,
    EntNearbyplaceFromJSON,
    EntNearbyplaceFromJSONTyped,
    EntNearbyplaceToJSON,
    EntQuantity,
    EntQuantityFromJSON,
    EntQuantityFromJSONTyped,
    EntQuantityToJSON,
    EntStaytype,
    EntStaytypeFromJSON,
    EntStaytypeFromJSONTyped,
    EntStaytypeToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntRoomdetailEdges
 */
export interface EntRoomdetailEdges {
    /**
     * 
     * @type {EntEmployee}
     * @memberof EntRoomdetailEdges
     */
    employee?: EntEmployee;
    /**
     * 
     * @type {EntEquipment}
     * @memberof EntRoomdetailEdges
     */
    equipments?: EntEquipment;
    /**
     * 
     * @type {EntFacilitie}
     * @memberof EntRoomdetailEdges
     */
    facilities?: EntFacilitie;
    /**
     * 
     * @type {EntNearbyplace}
     * @memberof EntRoomdetailEdges
     */
    nearbyplaces?: EntNearbyplace;
    /**
     * 
     * @type {EntQuantity}
     * @memberof EntRoomdetailEdges
     */
    quantity?: EntQuantity;
    /**
     * 
     * @type {EntLease}
     * @memberof EntRoomdetailEdges
     */
    roomdetails?: EntLease;
    /**
     * 
     * @type {EntStaytype}
     * @memberof EntRoomdetailEdges
     */
    staytype?: EntStaytype;
}

export function EntRoomdetailEdgesFromJSON(json: any): EntRoomdetailEdges {
    return EntRoomdetailEdgesFromJSONTyped(json, false);
}

export function EntRoomdetailEdgesFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntRoomdetailEdges {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'employee': !exists(json, 'employee') ? undefined : EntEmployeeFromJSON(json['employee']),
        'equipments': !exists(json, 'equipments') ? undefined : EntEquipmentFromJSON(json['equipments']),
        'facilities': !exists(json, 'facilities') ? undefined : EntFacilitieFromJSON(json['facilities']),
        'nearbyplaces': !exists(json, 'nearbyplaces') ? undefined : EntNearbyplaceFromJSON(json['nearbyplaces']),
        'quantity': !exists(json, 'quantity') ? undefined : EntQuantityFromJSON(json['quantity']),
        'roomdetails': !exists(json, 'roomdetails') ? undefined : EntLeaseFromJSON(json['roomdetails']),
        'staytype': !exists(json, 'staytype') ? undefined : EntStaytypeFromJSON(json['staytype']),
    };
}

export function EntRoomdetailEdgesToJSON(value?: EntRoomdetailEdges | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'employee': EntEmployeeToJSON(value.employee),
        'equipments': EntEquipmentToJSON(value.equipments),
        'facilities': EntFacilitieToJSON(value.facilities),
        'nearbyplaces': EntNearbyplaceToJSON(value.nearbyplaces),
        'quantity': EntQuantityToJSON(value.quantity),
        'roomdetails': EntLeaseToJSON(value.roomdetails),
        'staytype': EntStaytypeToJSON(value.staytype),
    };
}
