// plugin-welcome/src/api/models/EntRepairinvoice.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntRepairinvoiceEdges,
    EntRepairinvoiceEdgesFromJSON,
    EntRepairinvoiceEdgesFromJSONTyped,
    EntRepairinvoiceEdgesToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntRepairinvoice
 */
export interface EntRepairinvoice {
    /**
     * Bequipment holds the value of the "bequipment" field.
     * @type {string}
     * @memberof EntRepairinvoice
     */
    bequipment?: string;
    /**
     * 
     * @type {EntRepairinvoiceEdges}
     * @memberof EntRepairinvoice
     */
    edges?: EntRepairinvoiceEdges;
    /**
     * Emtell holds the value of the "emtell" field.
     * @type {string}
     * @memberof EntRepairinvoice
     */
    emtell?: string;
    /**
     * ID of the ent.
     * @type {number}
     * @memberof EntRepairinvoice
     */
    id?: number;
    /**
     * Num holds the value of the "num" field.
     * @type {number}
     * @memberof EntRepairinvoice
     */
    num?: number;
}

export function EntRepairinvoiceFromJSON(json: any): EntRepairinvoice {
    return EntRepairinvoiceFromJSONTyped(json, false);
}

export function EntRepairinvoiceFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntRepairinvoice {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'bequipment': !exists(json, 'bequipment') ? undefined : json['bequipment'],
        'edges': !exists(json, 'edges') ? undefined : EntRepairinvoiceEdgesFromJSON(json['edges']),
        'emtell': !exists(json, 'emtell') ? undefined : json['emtell'],
        'id': !exists(json, 'id') ? undefined : json['id'],
        'num': !exists(json, 'num') ? undefined : json['num'],
    };
}

export function EntRepairinvoiceToJSON(value?: EntRepairinvoice | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'bequipment': value.bequipment,
        'edges': EntRepairinvoiceEdgesToJSON(value.edges),
        'emtell': value.emtell,
        'id': value.id,
        'num': value.num,
    };
}
