// plugin-welcome/src/api/models/EntRentalstatus.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntRentalstatusEdges,
    EntRentalstatusEdgesFromJSON,
    EntRentalstatusEdgesFromJSONTyped,
    EntRentalstatusEdgesToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntRentalstatus
 */
export interface EntRentalstatus {
    /**
     * 
     * @type {EntRentalstatusEdges}
     * @memberof EntRentalstatus
     */
    edges?: EntRentalstatusEdges;
    /**
     * ID of the ent.
     * @type {number}
     * @memberof EntRentalstatus
     */
    id?: number;
    /**
     * Rentalstatus holds the value of the "rentalstatus" field.
     * @type {string}
     * @memberof EntRentalstatus
     */
    rentalstatus?: string;
}

export function EntRentalstatusFromJSON(json: any): EntRentalstatus {
    return EntRentalstatusFromJSONTyped(json, false);
}

export function EntRentalstatusFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntRentalstatus {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'edges': !exists(json, 'edges') ? undefined : EntRentalstatusEdgesFromJSON(json['edges']),
        'id': !exists(json, 'id') ? undefined : json['id'],
        'rentalstatus': !exists(json, 'rentalstatus') ? undefined : json['rentalstatus'],
    };
}

export function EntRentalstatusToJSON(value?: EntRentalstatus | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'edges': EntRentalstatusEdgesToJSON(value.edges),
        'id': value.id,
        'rentalstatus': value.rentalstatus,
    };
}
