// plugin-welcome/src/api/models/EntStatusdEdges.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntDeposit,
    EntDepositFromJSON,
    EntDepositFromJSONTyped,
    EntDepositToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntStatusdEdges
 */
export interface EntStatusdEdges {
    /**
     * Statusds holds the value of the statusds edge.
     * @type {Array<EntDeposit>}
     * @memberof EntStatusdEdges
     */
    statusds?: Array<EntDeposit>;
}

export function EntStatusdEdgesFromJSON(json: any): EntStatusdEdges {
    return EntStatusdEdgesFromJSONTyped(json, false);
}

export function EntStatusdEdgesFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntStatusdEdges {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'statusds': !exists(json, 'statusds') ? undefined : ((json['statusds'] as Array<any>).map(EntDepositFromJSON)),
    };
}

export function EntStatusdEdgesToJSON(value?: EntStatusdEdges | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'statusds': value.statusds === undefined ? undefined : ((value.statusds as Array<any>).map(EntDepositToJSON)),
    };
}
