// plugin-welcome/src/api/models/EntSituationEdges.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntBill,
    EntBillFromJSON,
    EntBillFromJSONTyped,
    EntBillToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntSituationEdges
 */
export interface EntSituationEdges {
    /**
     * Situations holds the value of the situations edge.
     * @type {Array<EntBill>}
     * @memberof EntSituationEdges
     */
    situations?: Array<EntBill>;
}

export function EntSituationEdgesFromJSON(json: any): EntSituationEdges {
    return EntSituationEdgesFromJSONTyped(json, false);
}

export function EntSituationEdgesFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntSituationEdges {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'situations': !exists(json, 'situations') ? undefined : ((json['situations'] as Array<any>).map(EntBillFromJSON)),
    };
}

export function EntSituationEdgesToJSON(value?: EntSituationEdges | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'situations': value.situations === undefined ? undefined : ((value.situations as Array<any>).map(EntBillToJSON)),
    };
}
