// plugin-welcome/src/api/models/EntSituation.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntSituationEdges,
    EntSituationEdgesFromJSON,
    EntSituationEdgesFromJSONTyped,
    EntSituationEdgesToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntSituation
 */
export interface EntSituation {
    /**
     * 
     * @type {EntSituationEdges}
     * @memberof EntSituation
     */
    edges?: EntSituationEdges;
    /**
     * ID of the ent.
     * @type {number}
     * @memberof EntSituation
     */
    id?: number;
    /**
     * Situationname holds the value of the "situationname" field.
     * @type {string}
     * @memberof EntSituation
     */
    situationname?: string;
}

export function EntSituationFromJSON(json: any): EntSituation {
    return EntSituationFromJSONTyped(json, false);
}

export function EntSituationFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntSituation {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'edges': !exists(json, 'edges') ? undefined : EntSituationEdgesFromJSON(json['edges']),
        'id': !exists(json, 'id') ? undefined : json['id'],
        'situationname': !exists(json, 'situationname') ? undefined : json['situationname'],
    };
}

export function EntSituationToJSON(value?: EntSituation | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'edges': EntSituationEdgesToJSON(value.edges),
        'id': value.id,
        'situationname': value.situationname,
    };
}
