// plugin-welcome/src/api/models/EntStatusd.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntStatusdEdges,
    EntStatusdEdgesFromJSON,
    EntStatusdEdgesFromJSONTyped,
    EntStatusdEdgesToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntStatusd
 */
export interface EntStatusd {
    /**
     * 
     * @type {EntStatusdEdges}
     * @memberof EntStatusd
     */
    edges?: EntStatusdEdges;
    /**
     * ID of the ent.
     * @type {number}
     * @memberof EntStatusd
     */
    id?: number;
    /**
     * Statusdname holds the value of the "statusdname" field.
     * @type {string}
     * @memberof EntStatusd
     */
    statusdname?: string;
}

export function EntStatusdFromJSON(json: any): EntStatusd {
    return EntStatusdFromJSONTyped(json, false);
}

export function EntStatusdFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntStatusd {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'edges': !exists(json, 'edges') ? undefined : EntStatusdEdgesFromJSON(json['edges']),
        'id': !exists(json, 'id') ? undefined : json['id'],
        'statusdname': !exists(json, 'statusdname') ? undefined : json['statusdname'],
    };
}

export function EntStatusdToJSON(value?: EntStatusd | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'edges': EntStatusdEdgesToJSON(value.edges),
        'id': value.id,
        'statusdname': value.statusdname,
    };
}
